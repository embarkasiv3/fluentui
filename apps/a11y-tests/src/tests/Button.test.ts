import { testComponent } from './testComponent';

testComponent({
  name: 'Button',
  url: '#/examples/button'
});
