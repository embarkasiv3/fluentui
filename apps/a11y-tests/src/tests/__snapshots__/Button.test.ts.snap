// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs Axe on Button 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"false\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-192\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "div:nth-child(1) > .css-192[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "http://localhost:4322/#/examples/button",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "div:nth-child(1) > .css-192[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"false\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-192\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "div:nth-child(2) > .css-192[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "http://localhost:4322/#/examples/button",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "div:nth-child(2) > .css-192[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"false\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-192\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "div:nth-child(3) > .css-192[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "http://localhost:4322/#/examples/button",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "div:nth-child(3) > .css-192[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"true\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-217\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".css-217",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "http://localhost:4322/#/examples/button",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".css-217",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"false\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-247\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".css-247",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "http://localhost:4322/#/examples/button",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".css-247",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
]
`;

exports[`runs Axe on Button 2`] = `"Found 5 a11y errors for Button. Please check 'Scan' tab in 'office-ui-fabric-react' on Azure DevOps"`;
